src/components/Modals/NewMessageModal.jsx

```diff
@@ -0,0 +1,220 @@
+// React Imports
+import React, { useState, useContext } from 'react';
+// Inrupt Library Imports
+import { useSession } from '@hooks';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CheckIcon from '@mui/icons-material/Check';
+import ClearIcon from '@mui/icons-material/Clear';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+// Utility Imports
+import { sendMessageTTL, getMessageTTL } from '../../utils';
+// Context Imports
+import { MessageContext, SignedInUserContext } from '../../contexts';
+
+/**
+ * messageFolderProps is an object that stores the props for the MessageFolder
+ * component
+ *
+ * @typedef newMessageModalProps
+ * @type {object}
+ * @property {() => void} closeForm - The function used to trigger NewMessageModal to
+ * close
+ * @memberof typedefs
+ */
+
+/**
+ * NewMessageModal Component - Component that allows user to write
+ *  a message to another user from their inbox
+ *
+ * @memberof Modals
+ * @name NewMessageModal
+ * @param {newMessageModalProps} Props - Props used for NewMessageModal
+ * @returns {React.JSX.Element} React component for NewMessageModal
+ */
+const NewMessageModal = ({ showModal, setShowModal, oldMessage = '' }) => {
+  const { session } = useSession();
+  const { outboxList, setOutboxList } = useContext(MessageContext);
+  const { podUrl } = useContext(SignedInUserContext);
+  const [originalMessage, setOriginalMessage] = useState(oldMessage.message);
+  const [replyMessage, setReplyMessage] = useState(false);
```

Comment:
If `replyMessage` will not be used, can be removed along with `setReplyMessage`.

Reply:
Should be done now. I'm still unsure what's causing the messages to be clicked when the reply version of the modal is open. It doesn't seem to happen when the regular, new message version is.

Reply:
I'm still seeing the line in Line 48 and Line 71. Perhaps the changes were not saved or committed?

Reply:
Oops. Should be done now.